crates/router/src/connector/cybersource/transformers.rs

```diff
@@ -1317,6 +1561,434 @@ impl<F>
     }
 }
 
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            CybersourceAuthSetupResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            CybersourceAuthSetupResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let connector_response_reference_id = Some(
+            item.response
+                .client_reference_information
+                .code
+                .unwrap_or(item.response.id.clone()),
+        );
+        Ok(Self {
+            status: enums::AttemptStatus::AuthenticationPending,
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
```

Comment:
What happens if connector throws 4xx error response in CybersourceAuthSetupResponse?

Reply:
4xx will be handled by `build_error_response` for 2xx errors have added a match